### PR TITLE
Linearize gpt_oss model and add separate example to qunatize it to w4a8

### PR DESCRIPTION
**Pull request by isharif168**
SUMMARY:
Add test to convert the gpt_oss model and then quantize model to w8a8

TEST PLAN:
Tested with gpt_oss model and run using vllm 

Sample output

=== Prompt 0 ===
<|start|>system<|message|>You are ChatGPT, a large language model trained by OpenAI.
Knowledge cutoff: 2024-06
Current date: 2025-12-04

Reasoning: medium

Valid channels: analysis, commentary, final. Channel must be included for every message.<|end|><|start|>developer<|message|># Instructions

You are a helpful assistant.

<|end|><|start|>user<|message|>Give 3 reasons to use AI.<|end|><|start|>assistant

--- Candidate 0 ---
analysisWe need to answer with three reasons to use AI. Should be short. Provide 3 reasons. Let's do it.assistantcommentaryHere are three reasons to use AI:

1. **Personalized learning** – AI can adapt to your learning style and pace, providing customized lessons that suit your learning style and progress.  
2. **Skill development** – AI can help you develop skills that are in demand, such as coding, data analysis, and data science.  
3. **Supportive design**  

We should provide a short answer.assistantfinal**Here are three reasons to use AI:**

1. **Personalized learning** – AI adapts to your learning style and pace, giving you personalized lessons that fit your needs.  
2. **Skill development** – AI helps you learn and practice skills that are in demand, like coding, data analysis, and data‑science skills.  
3. **Real‑world impact** – AI helps you practice real-world skills in real life, such as learning new skills and applying them in real life.
finish_reason: stop
num_tokens: 226

### CODE REVIEW

File: src/llmcompressor/modeling/gpt_oss.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+            )
+            parent._modules[child_name] = seq
+            to_delete.append(module)
+            print(f"[GPT-OSS] Patched {name} -> SequentialGPTOSSMoE (E={E}, inter={intermediate}, hidden={hidden})", flush=True)
+
+    for m in to_delete:
+        del m
+    if to_delete:
+        gc.collect()
+        try:
```
Comment (cfRod) [inline src/llmcompressor/modeling/gpt_oss.py:47]: Is this script expected to be used on GPUs? Has it been tested on GPU?

File: src/llmcompressor/modeling/gpt_oss.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            E, gH, g2i = gup.shape
+            Ed, dI, dH = dwn.shape
+            assert E == Ed and gH == dH
+            assert g2i % 2 == 0
+            intermediate = g2i // 2
+            hidden = gH
+
+            dtype = gup.dtype
+            parent, child_name = _get_parent_and_child(model, name)
+            top_k = int(max(1, min(_get_top_k(model.config) or 1, E)))
```
Comment (cfRod) [inline src/llmcompressor/modeling/gpt_oss.py:31]: Why cant topk just be defined as per the original definition https://github.com/huggingface/transformers/blob/390dca67e554b2b8f131064d4b6d991bf3ab3105/src/transformers/models/gpt_oss/modeling_gpt_oss.py#L154

File: src/llmcompressor/modeling/gpt_oss.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                    original_moe.experts.down_proj_bias[i]
+                )       # [H]
+
+
+    def forward(self, hidden_states):
+        B, T, H = hidden_states.shape
+        x = hidden_states.reshape(-1, H)
+
+        # Use the original router (it returns scores and indices already softmaxed over top-k)
+        router_scores, router_indices = self.router(x)   # scores: [tokens, E], indices: [tokens, k]
```
Comment (cfRod) [inline src/llmcompressor/modeling/gpt_oss.py:155]: The original definition of the router, returns router logits https://github.com/huggingface/transformers/blob/390dca67e554b2b8f131064d4b6d991bf3ab3105/src/transformers/models/gpt_oss/modeling_gpt_oss.py#L177 

is this right?

**Comment by aditew01** [comment]
@isharif168 can you please fix this: 
https://github.com/vllm-project/llm-compressor/pull/2091/checks?check_run_id=57154390621

**Comment by isharif168** [comment]
Hi @aditew01 
I am not sure how do I fix that , since the commits were from @shubhra 
Any ideas??
Thanks

File: src/llmcompressor/modeling/gpt_oss.py (hunk 1 above)
Comment (isharif168) [reply src/llmcompressor/modeling/gpt_oss.py:47]: not expected for GPU but it can support GPU as well

File: src/llmcompressor/modeling/gpt_oss.py (hunk 2 above)
Comment (isharif168) [reply src/llmcompressor/modeling/gpt_oss.py:31]: its the same with some housekeeping, get_top_k reads the same config parameters

**Comment by brian-dellabetta** [comment]
> @isharif168 can you please fix this: https://github.com/vllm-project/llm-compressor/pull/2091/checks?check_run_id=57154390621

@isharif168 @aditew01 don't worry about this, we can set this to resolved when we merge this

**Comment by aditew01** [comment]
@brian-dellabetta thanks for clarifying - sounds good to me!

**Comment by isharif168** [comment]
Closing as we dont require this change 
Thanks.
